fe/fe-core/src/main/java/com/starrocks/catalog/Column.java

```diff
@@ -335,13 +333,29 @@ public void checkSchemaChangeAllowed(Column other) throws DdlException {
             throw new DdlException("Can not change from nullable to non-nullable");
         }
 
-        if (this.getDefaultValue() == null) {
-            if (other.getDefaultValue() != null) {
-                throw new DdlException("Can not change default value");
+        DefaultValueType thisDefaultValueType = this.getDefaultValueType();
+        DefaultValueType otherDefaultValueType = other.getDefaultValueType();
+        // Adding a default value to a column without a default value is not supported
+        if (thisDefaultValueType != otherDefaultValueType) {
+            throw new DdlException(CAN_NOT_CHANGE_DEFAULT_VALUE);
+        }
+
+        // default value should same
+        if (thisDefaultValueType == DefaultValueType.VARY) {
+            if (!this.getDefaultExpr().getExpr().equals(other.getDefaultExpr().getExpr())) {
+                throw new DdlException(CAN_NOT_CHANGE_DEFAULT_VALUE);
             }
-        } else {
-            if (!this.getDefaultValue().equals(other.getDefaultValue())) {
-                throw new DdlException("Can not change default value");
+        } else if (this.getDefaultValueType() == DefaultValueType.CONST) {
+            if (this.getDefaultValue() != null && other.getDefaultValue() != null) {
+                if (!this.getDefaultValue().equals(other.getDefaultValue())) {
+                    throw new DdlException(CAN_NOT_CHANGE_DEFAULT_VALUE);
+                }
+            } else if (this.getDefaultExpr() != null && other.getDefaultExpr() != null) {
```

Comment:
This is because CONST has 2 cases, one is DEFAULT "1" , this will make this.getDefaultExpr == null
and the other is DEFAULT CURRENT_TIMESTAMP, this will make this.getDefaultExpr != null